lightning/src/ln/channel.rs

```diff
@@ -6086,16 +6088,34 @@ where
 
 #[cfg(splicing)]
 macro_rules! promote_splice_funding {
-	($self: expr, $funding: expr) => {
+	($self: expr, $funding: expr) => {{
+		let prev_funding_txid = $self.funding.get_funding_txid();
 		if let Some(scid) = $self.funding.short_channel_id {
 			$self.context.historical_scids.push(scid);
 		}
 		core::mem::swap(&mut $self.funding, $funding);
 		$self.interactive_tx_signing_session = None;
 		$self.pending_splice = None;
-		$self.pending_funding.clear();
 		$self.context.announcement_sigs_state = AnnouncementSigsState::NotSent;
-	};
+
+		let discarded_funding = $self
+			.pending_funding
+			.drain(..)
+			.filter(|funding| funding.get_funding_txid() != prev_funding_txid)
```

Comment:
Could you add a comment here about the above `swap` as was already done in `channelmonitor.rs`?